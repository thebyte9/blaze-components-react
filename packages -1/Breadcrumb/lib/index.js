"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var Breadcrumb = function (_a) {
    var children = _a.children;
    return (react_1["default"].createElement("ul", { className: "breadcrumb" }, react_1["default"].Children.map(children, function (child) {
        return react_1["default"].createElement("li", { key: child.symbol, className: "breadcrumb__item" }, child);
    })));
};
Breadcrumb.defaultProps = {
    children: "Missing breadcrumb content"
};
exports["default"] = Breadcrumb;
