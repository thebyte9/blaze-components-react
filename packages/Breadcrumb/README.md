
## Description

Breadcrumbs allow users to make selections from a range of values.

## Usage

* Simple

```js
<Breadcrumb>
  <a href="/">First</a>
  <a href="/">Second</a>
  <a href="/">
    Third - long text will be truncated for a better user experience
  </a>
</Breadcrumb>
```

## API

##### Breadcrumb can receive a number of `props` as follow:


| NAME   | TYPE | DEFAULT | 
| :---  | :---:  | :---: |
| children | A single or array of nodes| Missing breadcrumb content |

