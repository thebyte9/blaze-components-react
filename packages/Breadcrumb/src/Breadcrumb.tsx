import React, { FunctionComponent } from "react";

const Breadcrumb: FunctionComponent<any> = ({ children }) => (
  <ul className="breadcrumb">
    {React.Children.map(
      children,
      (child): React.ReactElement =>
        <li key={child.symbol} className="breadcrumb__item">{child}</li>
    )}
  </ul >
);

Breadcrumb.defaultProps = {
  children: "Missing breadcrumb content"
};

export default Breadcrumb;
