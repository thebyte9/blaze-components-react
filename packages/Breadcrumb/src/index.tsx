import React, { FunctionComponent } from "react";

const Breadcrumb: FunctionComponent<any> = ({ children }) => (
  <ul className="breadcrumb" data-testid="breadcrumb">
    {React.Children.map(
      children,
      (child: any): JSX.Element =>
        <li key={child.symbol} className="breadcrumb__item">{child}</li>
    )}
  </ul >
);

Breadcrumb.defaultProps = {
  children: "Missing breadcrumb content"
};
export default Breadcrumb;
