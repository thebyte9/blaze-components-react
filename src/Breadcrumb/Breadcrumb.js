import React from 'react';
import PropTypes from 'prop-types';

const Breadcrumb = ({
  children
}) => (
  <ul className="breadcrumb">
    {children.length
      ? children.map(child => <li key={Math.random()} className="breadcrumb__item">{child}</li>)
      : <li className="breadcrumb__item">{children}</li>
    }
  </ul>
);

Breadcrumb.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

Breadcrumb.defaultProps = {
  children: 'No content'
};

export default Breadcrumb;
