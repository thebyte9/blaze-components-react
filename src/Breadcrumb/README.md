
## Description

Breadcrumbs allow users to make selections from a range of values.

## Usage

* Simple

```js
<Breadcrumb>
  <a href="/">First</a>
  <a href="/">Second</a>
  <a href="/">Third - long text will be truncated for a better user experience</a>
</Breadcrumb>
```

## API

##### Textarea can receive a number of `props` as follow:


| NAME   | TYPE | DEFAULT | 
| :---  | :---:  | :---: |
| label | String | '' | 
| value | String | ''  | 
| required | Boolean | false | 
| limit | Number | 0      |
| children | A single or array of nodes| Missing breadcrumb content |

