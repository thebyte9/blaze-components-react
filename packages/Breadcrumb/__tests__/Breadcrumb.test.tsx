import { shallow } from 'enzyme';
import expect from 'expect';
import React from 'react';
import Breadcrumb from '../src';

const defaultProps = (override = {}) => ({
  children: [
    <a href="#first">First</a>,
    <a href="#Second">Second</a>,
    <a href="#Third">Third - long text will be truncated for a better user experience</a>
  ],
  ...override
});

const wrapper = (props: any) => shallow(<Breadcrumb {...props} />);

describe('Breadcrumb component', () => {
  test('should be defined and renders correctly (snapshot)', () => {
    const _wrapper = wrapper(defaultProps());

    expect(_wrapper).toBeDefined();
    expect(_wrapper).toMatchSnapshot();
  });

  test('should renders correctly with one children', () => {
    const children = <a href="#first">First</a>;
    const _wrapper = wrapper(defaultProps({ children }));

    expect(_wrapper).toBeDefined();
    expect(_wrapper).toMatchSnapshot();
  });
});
