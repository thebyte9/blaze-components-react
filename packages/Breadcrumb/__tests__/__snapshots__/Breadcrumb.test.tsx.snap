// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component should be defined and renders correctly (snapshot) 1`] = `
<ul
  className="breadcrumb"
>
  <li
    className="breadcrumb__item"
    key=".0"
  >
    <a
      href="#first"
    >
      First
    </a>
  </li>
  <li
    className="breadcrumb__item"
    key=".1"
  >
    <a
      href="#Second"
    >
      Second
    </a>
  </li>
  <li
    className="breadcrumb__item"
    key=".2"
  >
    <a
      href="#Third"
    >
      Third - long text will be truncated for a better user experience
    </a>
  </li>
</ul>
`;

exports[`Breadcrumb component should renders correctly with one children 1`] = `
<ul
  className="breadcrumb"
>
  <li
    className="breadcrumb__item"
    key=".0"
  >
    <a
      href="#first"
    >
      First
    </a>
  </li>
</ul>
`;
